sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2556,6 +2556,91 @@ case class MakeDate(
     copy(year = newFirst, month = newSecond, day = newThird)
 }
 
+// scalastyle:off line.size.limit
+@ExpressionDescription(
+  usage = "_FUNC_(hour, minute, second) - Create time from hour, minute and second fields. If the configuration `spark.sql.ansi.enabled` is false, the function returns NULL on invalid inputs. Otherwise, it will throw an error instead.",
```

Comment:
I believe we don't need the non-ANSI behavior in new functions/expression. Let's follow the ANSI SQL standard, and just raise an error.

If users need NULLs on invalid inputs, they can invoke `try_make_time` which we introduce later.

Reply:
I updated this, but needed to create a new error code `DATETIME_FIELD_OUT_OF_BOUNDS_ANSI_ONLY` since the existing one suggests disabling ansi behavior.

I decided to create the new code to minimize changes and the change in a public facing error code could be considered a breaking changed. Absent those concerns I would have liked to make the new error code `DATETIME_FIELD_OUT_OF_BOUNDS` and rename the existing one to `DATETIME_FIELD_OUT_OF_BOUNDS_ANSI_OPTIONAL`.